packages/gatsby-theme-store/src/sdk/region/index.tsx

```diff
@@ -0,0 +1,66 @@
+/** @jsx jsx */
+import { jsx } from '@vtex/store-ui'
+import type { FC } from 'react'
+import React, { useMemo, useState } from 'react'
+
+export interface RegionContextType {
+  postalCode?: null | string
+  regionId?: null | string
+  // TODO: fix weird string | null | undefined types
+  setPostalCode: (value: string | null | undefined) => void
+  setRegionId: (value: string | null | undefined) => void
+}
+export const RegionContext = React.createContext<RegionContextType>({
+  postalCode: null,
+  regionId: null,
+  setPostalCode: (_: string | null | undefined) => {},
+  setRegionId: (_: string | null | undefined) => {},
+})
+
+export const RegionProvider: FC = ({ children }) => {
+  const [postalCode, setPostalCode] = useState<string | null | undefined>(
+    () => {
+      const savedPostalCode = window?.localStorage?.getItem('vtex:postalCode')
```

Comment:
Just a comment, we should avoid using localStorage where we can because of the synchronous access.

https://web.dev/storage-for-the-web/

I think we should create a module to abstract this storage so we can change it to IndexedDB later. In this module we can use `[idb](https://github.com/jakearchibald/idb)`.

cc @tlgimenes

Reply:
it doesn't really make a difference in this case, but we can look into it in the future

Reply:
also, you'd have to take into account the overhead that creating, maintaining, loading, and initializing a module introduces and compare with the cost that using `localStorage` has, which for our use case is very low.